segmentation/utils/spatial_analysis.py

```diff
@@ -56,200 +55,203 @@ def calculate_channel_spatial_enrichment(dist_matrices, marker_thresholds, all_d
+    if not (list(marker_thresholds.iloc[:, 0]) == channel_titles).any():
         raise ValueError("Threshold Markers do not match markers in Expression Matrix")
 
-    # Create stats Xarray with the dimensions (points, stats variables, number of markers, number of markers)
-    stats_raw_data = np.zeros((num_fovs, 7, marker_num, marker_num))
-    coords = [fovs, ["z", "muhat", "sigmahat", "p_pos", "p_neg", "h", "p_adj"], marker_titles,
-              marker_titles]
-    dims = ["points", "stats", "marker1", "marker2"]
+    # Create stats Xarray with the dimensions (fovs, stats variables, number of channels, number of channels)
+    stats_raw_data = np.zeros((num_fovs, 7, channel_num, channel_num))
+    coords = [included_fovs, ["z", "muhat", "sigmahat", "p_pos", "p_neg", "h", "p_adj"], channel_titles,
+              channel_titles]
+    dims = ["fovs", "stats", "marker1", "marker2"]
     stats = xr.DataArray(stats_raw_data, coords=coords, dims=dims)
 
     # Subsetting threshold matrix to only include column with threshold values
     thresh_vec = marker_thresholds.iloc[:, 1]
 
-    for i in range(0, len(fovs)):
-        # Subsetting expression matrix to only include patients with correct label
-        patient_ids = all_data[fov_col] == fovs[i]
-        fov_data = all_data[patient_ids]
-        # Patients with correct label, and only columns of markers
-        fov_channel_data = data_markers[patient_ids]
+    for i in range(0, len(included_fovs)):
+        # Subsetting expression matrix to only include patients with correct fov label
+        current_fov_idx = all_data[fov_col] == included_fovs[i]
+        current_fov_data = all_data[current_fov_idx]
+        # Patients with correct label, and only columns of channel markers
+        current_fov_channel_data = all_channel_data[current_fov_idx]
 
         # Subset the distance matrix dictionary to only include the distance matrix for the correct point
-        dist_matrix = dist_matrices[str(fovs[i])]
+        dist_matrix = dist_matrices_dict[str(included_fovs[i])]
 
         # Get close_num and close_num_rand
-        close_num, marker_nums = spatial_analysis_utils.compute_close_cell_num(
-            dist_mat=dist_matrix, dist_lim=100, num=marker_num, analysis_type="Channel",
-            fov_data=fov_data, fov_channel_data=fov_channel_data, thresh_vec=thresh_vec)
+
+        close_num, marker1_num, marker2_num = spatial_analysis_utils.compute_close_cell_num(
+            dist_mat=dist_matrix, dist_lim=100, num=channel_num, analysis_type="channel",
+            current_fov_data=current_fov_data, current_fov_channel_data=current_fov_channel_data,
+            thresh_vec=thresh_vec)
 
         close_num_rand = spatial_analysis_utils.compute_close_cell_num_random(
-            marker_nums, dist_matrix, dist_lim, bootstrap_num)
+            marker1_num, marker2_num, dist_matrix, channel_num, dist_lim, bootstrap_num)
```

Comment:
change `close_num, marker1_num, marker2_num` to `close_num, marker_nums`on line 94 and revert change on line 100 (no marker2_num, channel_num, and marker1_num -> marker_nums)